packages/slider/src/HandleController.ts

```diff
@@ -0,0 +1,540 @@
+
+    private dispatchChangeEvent(
+        input: HTMLInputElement,
+        handle: SliderHandle
+    ): void {
+        input.valueAsNumber = handle.value;
+
+        const changeEvent = new Event('change', {
+            bubbles: true,
+            composed: true,
+        });
+
+        handle.dispatchEvent(changeEvent);
+    }
+
+    /**
+     * Returns the value under the cursor
+     * @param: PointerEvent on slider
+     * @return: Slider value that correlates to the position under the pointer
+     */
+    private calculateHandlePosition(
+        event: PointerEvent | MouseEvent,
+        model: ModelValue
+    ): number {
+        const rect = this.boundingClientRect;
+        const minOffset = rect.left;
+        const offset = event.clientX;
+        const size = rect.width;
+
+        const normalized = (offset - minOffset) / size;
+        const value = model.normalization.fromNormalized(
+            normalized,
+            model.range.min,
+            model.range.max
+        );
+
+        /* c8 ignore next */
+        return this.host.isLTR ? value : model.range.max - value;
+    }
+
+    public renderHandle(
+        model: ModelValue,
+        index: number,
+        zIndex: number
+    ): TemplateResult {
+        const isActive = this.activeHandle === model.name;
+        const classes = {
+            handle: true,
+            dragging: this.draggingHandle?.handleName === model.name,
+            'handle-highlight': model.highlight,
+        };
+        const style = {
+            [this.host.isLTR ? 'left' : 'right']: `${
+                model.normalizedValue * 100
+            }%`,
+            'z-index': zIndex.toString(),
+            // Allow setting background per-handle
+            'background-color': `var(--spectrum-slider-handle-background-color-${index}, var(--spectrum-slider-handle-default-background-color))`,
+            'border-color': `var(--spectrum-slider-handle-border-color-${index}, var(-spectrum-slider-handle-default-border-color))`,
+        };
+        return html`
+            <div
+                class=${classMap(classes)}
+                name=${model.name}
+                style=${styleMap(style)}
+                @manage=${streamingListener(
+                    {
+                        type: 'pointerdown',
+                        fn: (event) => this.handlePointerdown(event, model),
+                    },
+                    {
+                        type: 'pointermove',
+                        fn: (event) => this.handlePointermove(event, model),
+                    },
+                    {
+                        type: ['pointerup', 'pointercancel'],
+                        fn: (event) => this.handlePointerup(event, model),
+                    }
+                )}
+                role="presentation"
+            >
+                <input
+                    type="range"
+                    class="input"
+                    id=${ifDefined(isActive ? 'input' : undefined)}
```

Comment:
We likely should switch from the current `<label for="id"/><output for="id"/><input id="id" />` style of labeling to a `<label id="id1" /><label id="id1" /><input aria-labelledby="id1 id2" />` style so that any number of inputs can reference the `label` and `output` element at the same time.

Reply:
I am a bit confused. Your suggestion above seems to show multiple labels for one input. Whereas I we have multiple inputs for one label. Am I missing something? 😕

Reply:
Ok, that looks similar to what I did based on React Spectrum